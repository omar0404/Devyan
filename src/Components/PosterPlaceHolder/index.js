import PosterPlaceHodler from './PosterPlaceHolder';
export default PosterPlaceHodler;
