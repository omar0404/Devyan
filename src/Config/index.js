export const POSTER_BASE_URL = 'http://image.tmdb.org/t/p/w500';
