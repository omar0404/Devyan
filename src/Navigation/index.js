import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HomeScreens from './HomeScreens';

const RootNavigation = () => (
  <NavigationContainer>
    <HomeScreens />
  </NavigationContainer>
);
export default RootNavigation;
