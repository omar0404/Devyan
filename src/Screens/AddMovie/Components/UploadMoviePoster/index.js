import UploadMoviePoster from './UploadMoviePoster';
export default UploadMoviePoster;
