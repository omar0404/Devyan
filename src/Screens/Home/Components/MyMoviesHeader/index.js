import MyMoviesHeader from './MyMoviesHeader';
export default MyMoviesHeader;
