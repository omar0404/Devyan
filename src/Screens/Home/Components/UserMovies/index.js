import UserMovies from './UserMovies';
export default UserMovies;
