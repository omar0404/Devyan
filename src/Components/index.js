import Section from './Section';
import PosterPlaceHolder from './PosterPlaceHolder';
import MovieCard from './MovieCard';
export {MovieCard, Section, PosterPlaceHolder};
