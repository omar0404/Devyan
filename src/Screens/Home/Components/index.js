import MyMovies from './MyMovies';
import AllMovies from './AllMovies';
import UserMovies from './UserMovies';
import MyMoviesHeader from './MyMoviesHeader';
export {MyMovies, AllMovies, UserMovies, MyMoviesHeader};
