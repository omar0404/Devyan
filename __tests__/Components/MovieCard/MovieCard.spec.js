import 'react-native';
import React from 'react';
import {MovieCard} from 'MoviesApp/src/Components';

import renderer from 'react-test-renderer';
import {POSTER_BASE_URL} from 'MoviesApp/src/Config';

describe('Movie Card Componennt', () => {
  it('renders correctly', () => {
    const MovieCardComponent = renderer.create(<MovieCard />).toJSON();
    expect(MovieCardComponent).toMatchSnapshot();
  });

  it('renders movie poster', () => {
    const props = {
      movie: {posterUri: POSTER_BASE_URL},
    };
    const MovieCardComponent = renderer
      .create(<MovieCard {...props} />)
      .toJSON();
    expect(MovieCardComponent).toMatchSnapshot();
  });
});
