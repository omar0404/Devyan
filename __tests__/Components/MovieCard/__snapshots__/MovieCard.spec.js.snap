// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card Componennt renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 200,
      "marginRight": 20,
      "opacity": 1,
      "width": 200,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 2,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "gray",
            "fontSize": 50,
          },
          undefined,
          Object {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "white",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "flex": 1,
          }
        }
      >
        <Text>
           
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#991F25",
              "flex": 1,
              "fontWeight": "bold",
              "textAlign": "right",
            }
          }
        />
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "gray",
          "marginTop": 12,
        }
      }
    />
  </View>
</View>
`;

exports[`Movie Card Componennt renders movie poster 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "height": 200,
      "marginRight": 20,
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "http://image.tmdb.org/t/p/w500",
      }
    }
    style={
      Object {
        "flex": 2,
        "height": null,
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "marginTop": 12,
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "white",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "rgb(255, 255, 255)",
            "flex": 1,
          }
        }
      >
        <Text>
           
        </Text>
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#991F25",
              "flex": 1,
              "fontWeight": "bold",
              "textAlign": "right",
            }
          }
        />
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 30,
            },
            undefined,
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "gray",
          "marginTop": 12,
        }
      }
    />
  </View>
</View>
`;
